src/passes/Print.cpp

```diff
@@ -3185,32 +3186,41 @@ void PrintSExpression::visitImportedTag(Tag* curr) {
   emitImportHeader(curr);
   o << "(tag ";
   curr->name.print(o);
-  if (curr->sig.params != Type::none) {
-    o << maybeSpace;
-    printParamType(curr->sig.params);
-  }
-  if (curr->sig.results != Type::none) {
-    o << maybeSpace;
-    printResultType(curr->sig.results);
-  }
-  o << "))";
-  o << maybeNewLine;
+  o << maybeSpace;
+  printTagType(curr->type);
+  o << "))" << maybeNewLine;
 }
 
 void PrintSExpression::visitDefinedTag(Tag* curr) {
   doIndent(o, indent);
   o << '(';
   printMedium(o, "tag ");
   curr->name.print(o);
-  if (curr->sig.params != Type::none) {
-    o << maybeSpace;
-    printParamType(curr->sig.params);
+  o << maybeSpace;
+  printTagType(curr->type);
+  o << ')' << maybeNewLine;
+}
+
+void PrintSExpression::printTagType(HeapType type) {
```

Comment:
Looks like `handleSignature` (used for Function) is similar - can we share this code perhaps?

Reply:
I'll take another look at it. At first glance the existence of parameter names for functions made it seem writing separate code.

Reply:
Yep, I think it's more trouble than it's worth right now.